File: NOVEMBER-2025/16-11-2025/1513.rb
def num_sub(s)
  left = 0
  count = 0
  res = 0
  mod = 1_000_000_007
  s.each_char.with_index do |char, right|
    if char == '1'
      count += 1
      res = (res + count) % mod
    else
      count = 0
    end
  end
  res
end

File: NOVEMBER-2025/16-11-2025/1513.go
func numSub(s string) int {
    res := 0
    const MODULO = 1000000007
    for i := 0; i < len(s); i++ {
        if s[i] == '0' {
            continue
        }
        j := i
        for ;j < len(s) && s[j] == '1';j++ {}
        val := (j-i)
        res += (val*(val+1)/2)
        res = res%MODULO
        i=j
    }
    return res
}